src/main/java/com/synopsys/integration/detect/tool/signaturescanner/operation/SignatureScanResult.java

```diff
@@ -45,16 +46,29 @@ public Set<String> parseScanIds() {
             for (String scanId : getScans().values()) {
                 // This can happen if we get a NOT_EXECUTED scan if the scanner decides not to
                 // run the scan
-                if (scanId != null) {
+                if (scanId != null && !checkInvalidScanID(scanId)) {
                     ids.add(scanId);
                 }
             }
-        } else if (getScanId() != null) {
+        } else if (getScanId() != null && !checkInvalidScanID(getScanId()))  {
             // If we are using an older version of the signature scanner, prior to 2023.1.0,
             // the scans field will not exist. Fallback to seeing if we have a high level scan ID
             ids.add(getScanId());
         }
         
         return ids;
     }
+    private boolean checkInvalidScanID(String scanId) {
+        // if BlackDuck returns an invalid scanID (containing only zeros) or an invalid UUID
+        // this method will guard.
+
+        Pattern validUUIDRegex = Pattern
+                .compile("^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$");
+        Pattern validScanIdRegex = Pattern.compile("^0+-0+-0+-0+-0+$");
```

Comment:
What will happen if another invalid pattern will come in future? Shall we write another pattern and add another else block below?
Won't it be better if we can introduce a list of invalid pattern and loop through that for validation?

Reply:
I think, we are only checking for invalid scanID, provided it is a valid UUID. Though, **UUID that contains all zero is a valid UUID**, but in our case it is not **valid scanID**. Hence, we won't process the request containing invalid scanID and this method ensures that.